analyses/doublet-detection/scripts/01a_run-scdblfinder.R

```diff
@@ -105,35 +140,53 @@ option_list <- list(
 )
 opts <- parse_args(OptionParser(option_list = option_list))
 
-# Check input arguments
-if (is.null(opts$dataset_name)) {
-  stop("Must provide a dataset name with --dataset_name.")
-}
-if (is.null(opts$data_dir)) {
-  stop("Must provide an input path for the SCE file with --data_dir.")
+# Check input arguments and set up files, directories ------
+if (!file.exists(opts$input_sce_file)) {
+    glue::glue("Could not find input SCE file, expected at: `{opts$input_sce_file}`.")
 }
 if (is.null(opts$results_dir)) {
   stop("Must provide an output path for results with --results_dir.")
 }
 
+output_tsv_file <- file.path(
+  opts$results_dir,
+  stringr::str_replace(
+    basename(opts$input_sce_file),
+    ".rds",
+    "_scdblfinder.tsv"
+  )
+)
+
 fs::dir_create(opts$results_dir)
 set.seed(opts$random_seed)
 
-
 # Detect doublets and export TSV file with inferences -----
-input_sce_file <- file.path(opts$data_dir, glue::glue("{opts$dataset_name}_sce.rds"))
-if (!file.exists(input_sce_file)) {
-  stop(
-    glue::glue("Could not find input file, expected at: `{input_sce_file}`.")
+
+# Check the number of cells to see if scDblFinder can be run
+cell_threshold <- 10
+
+sce <- readRDS(opts$input_sce_file)
+ncells <- ncol(sce)
+
+if (ncells < cell_threshold) {
+  warning(
+    glue::glue(
+      "The provided SingleCellExperiment object only has {ncells} droplets, which is fewer than {cell_threshold} so `scDblFinder` will not be run.
+       An output TSV file will still be produced, but it will be populated with `NA` values."
+    )
+  )
+  export_na_tsv(
+    colnames(sce),
+    output_tsv_file
```

Comment:
I think you could remove your function all together and replace with: 

```suggestion
  data.frame(
    barcodes = colnames(sce),
    score = NA, 
    class = NA
  ) |> 
  readr::write_tsv(output_tsv_file)
```